classes/cliTool/ConvertLogFile.php

```diff
@@ -0,0 +1,1012 @@
+     * based on the passed offset.
+     */
+    protected static function getUrlComponents(string $urlInfo, bool $isPathInfo, int $offset, string $varName = ''): mixed
+    {
+        $component = null;
+
+        $isArrayComponent = false;
+        if ($varName == 'path') {
+            $isArrayComponent = true;
+        }
+        if ($isPathInfo) {
+            $application = Application::get();
+            $contextDepth = $application->getContextDepth();
+
+            $vars = explode('/', trim($urlInfo, '/'));
+            if (count($vars) > $contextDepth + $offset) {
+                if ($isArrayComponent) {
+                    $component = array_slice($vars, $contextDepth + $offset);
+                } else {
+                    $component = $vars[$contextDepth + $offset];
+                }
+            }
+        } else {
+            parse_str((string) parse_url($urlInfo, PHP_URL_QUERY), $userVarsFromUrl);
+            $component = $userVarsFromUrl[$varName] ?? null;
+        }
+
+        if ($isArrayComponent) {
+            if (empty($component)) {
+                $component = [];
+            } elseif (!is_array($component)) {
+                $component = [$component];
+            }
+        }
+
+        return $component;
+    }
+
+    /**
+     * Set assoc type and IDs from the passed page, operation and arguments.
+     */
+    protected function setAssoc(int $assocType, string $op, array $args, array &$newEntry): void
+    {
+        $application = Application::get();
+        $applicationName = $application->getName();
+        switch ($applicationName) {
+            case 'ojs2':
+                $this->setOJSAssoc($assocType, $args, $newEntry);
+                break;
+            case 'omp':
+                $this->setOMPAssoc($assocType, $args, $newEntry);
+                break;
+            case 'ops':
+                $this->setOPSAssoc($assocType, $args, $newEntry);
+                break;
+            default:
+                throw new Exception('Unrecognized application name!');
+        }
+    }
+
+    /**
+     * Set assoc type and IDs from the passed page, operation and
+     * arguments specific to OJS.
+     */
+    protected function setOJSAssoc(int $assocType, array $args, array &$newEntry): void
+    {
+        switch ($assocType) {
+            case Application::getContextAssocType():
+                // $newEntry['contextId'] has already been set
+                $newEntry['assocType'] = $assocType;
+                $newEntry['submissionId'] = null;
+                $newEntry['representationId'] = null;
+                $newEntry['submissionFileId'] = null;
+                $newEntry['fileType'] = null;
+                $newEntry['issueId'] = null;
+                $newEntry['issueGalleyId'] = null;
+                break;
+
+            case Application::ASSOC_TYPE_SUBMISSION:
+                if (!isset($args[0])) {
+                    echo "Missing URL parameter.\n";
```

Comment:
The messages here should include enough context to be able to identify and fix missing data.

Reply:
I forgot to say that after such a message (when the assocType is not set) this message will be displayed too:
"The URL {$entryData['url']} in the line number {$lineNumber} was not considered.\n"
(s. https://github.com/bozana/pkp-lib/blob/6849/classes/cliTool/ConvertLogFile.php#L186):
Thus, now I think there is sufficient information there...